crates/sui-adapter/src/execution_engine.rs

```diff
@@ -586,18 +571,23 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
         temporary_store.drop_writes();
         // Must reset the storage rebate since we are re-executing.
         gas_status.reset_storage_cost_and_rebate();
-        let advance_epoch_safe_mode_pt =
-            construct_advance_epoch_safe_mode_pt(&params, protocol_config)?;
-        programmable_transactions::execution::execute::<_, execution_mode::System>(
-            protocol_config,
-            move_vm,
-            temporary_store,
-            tx_ctx,
-            gas_status,
-            None,
-            advance_epoch_safe_mode_pt,
-        )
-        .expect("Advance epoch with safe mode must succeed");
+
+        if protocol_config.get_advance_epoch_start_time_in_safe_mode() {
```

Comment:
I see this is where we are piggybacking. And we can only do this because we haven't entered safe mode in testnet since we added `get_advance_epoch_start_time_in_safe_mode`, right? What if we indeed enter safe mode between now and when this code is deployed?

Reply:
It should still be ok? Given that the Move version and Rust version are identical

Reply:
Good point.